INTDB-301-DOC: Move Federated Docs to separate branch (#747)
mongodb/terraform-provider-mongodbatlas: website/docs/r/federated_settings_identity_provider.html.markdown

---
layout: "mongodbatlas"

> page_title: "MongoDB Atlas: mongodbatlas_cloud_federated_settings_identity_provider"

Reviewer: ```suggestion
page_title: "MongoDB Atlas: mongodbatlas_federated_settings_identity_provider"
```

> sidebar_current: "docs-mongodbatlas-cloud-federated-settings-identity-provider"

Reviewer: ```suggestion
sidebar_current: "docs-mongodbatlas-federated-settings-identity-provider"
```

description: |-

>     Provides an Federated Settings Identity Provider Resource.

Reviewer: ```suggestioni
    Provides a federated settings identity provider resource.
```

Reviewer: Missed

---

> # mongodbatlas_cloud_federated_settings_identity_provider

Reviewer: ```suggestion
# mongodbatlas_federated_settings_identity_provider
```

> `mongodbatlas_cloud_federated_settings_identity_provider` provides an Atlas Cloud Federated Settings Identity Provider resource provides a subset of settings to be maintained post import of the existing resource.

Reviewer: ```suggestion
`mongodbatlas_federated_settings_identity_provider` provides an Atlas federated settings identity provider resource which provides a subset of settings to be managed post import of the existing resource.
```

> ## Example Usage

Reviewer: We need a very clear note to users they have to import this first or they will miss it.

```suggestion

~> **IMPORTANT** You **MUST** import this resource before you can manage it with this provider. 

## Example Usage
```

Reviewer: Still need to add these.

Reviewer: @martinstibbe not fixed.

Author: Added to identity provider resource

```terraform

> resource "mongodbatlas_cloud_federated_settings_identity_provider" "identity_provider" {

Reviewer: ```suggestion
resource "mongodbatlas_federated_settings_identity_provider" "identity_provider" {
```

  federation_settings_id     = "627a9687f7f7f7f774de306f14"
  name = "mongodb_federation_test"
  associated_domains           = ["yourdomain.com"]
  sso_debug_enabled = true
  status = "ACTIVE"
  sso_url = "https://mysso.oktapreview.com/app/mysso_terraformtestsso/exk17q7f7f7f7f50h8/sso/saml"
  issuer_uri = "http://www.okta.com/exk17q7f7f7f7fp50h8"
  request_binding = "HTTP-POST"
  response_signature_algorithm = "SHA-256"
}
```

## Argument Reference

* `federation_settings_id` - (Required) Unique 24-hexadecimal digit string that identifies the federated authentication configuration.
* `name` - (Required) Human-readable label that identifies the identity provider.
* `associated_domains` - (Required) List that contains the domains associated with the identity provider.
* `sso_debug_enabled` - (Required) Flag that indicates whether the identity provider has SSO debug enabled.
* `status`- (Required) String enum that indicates whether the identity provider is active or not. Accepted values are ACTIVE or INACTIVE.

> * `issuer_uri` - (Required) Identifier for the issuer of the SAML Assertion.

Reviewer: ```suggestion
* `issuer_uri` - (Required) Unique string that identifies the issuer of the SAML Assertion.

Reviewer: @martinstibbe not fixed.

Author: Fixed

> * `sso_url` - (Required) URL of the receiver of the SAML AuthNRequest.

Reviewer: The suggestion is what the source docs has, if you feel yours is more descriptive I'm okay with it too.
```suggestion
* `sso_url` - (Required) Unique string that identifies the intended audience of the SAML assertion.
```

Reviewer: @martinstibbe not fixed.

Author: Fixed

> * `request_binding` - (Required) SAML Authentication Request Protocol binding used to send the AuthNRequest. Atlas supports the following binding values:

Reviewer: Same as above, source is different but if you feel edit better captures I'm okay to change.
```suggestion
* `request_binding` - (Required) SAML Authentication Request Protocol HTTP method binding (POST or REDIRECT) that Federated Authentication uses to send the authentication request. Atlas supports the following binding values:
```

Reviewer: @martinstibbe not fixed.

Author: Fixed

    - HTTP POST
    - HTTP REDIRECT
* `response_signature_algorithm` - (Required) Algorithm used to encrypt the IdP signature. Atlas supports the following signature algorithm values:

>     - SHA-1

Reviewer: If we just left this SHA-1 the questions would come so fast it would shock you.
```suggestion
* `response_signature_algorithm` - (Required) Signature algorithm that Federated Authentication uses to encrypt the identity provider signature.  Valid values include SHA-1 and SHA-256.
```

Reviewer: @martinstibbe not fixed.

Author: Fixed

## Attributes Reference

In addition to all arguments above, the following attributes are exported:

### FederatedSettingsIdentityProvider

* `idp_id` - Unique 20-hexadecimal digit string that identifies the IdP.

## Import

> Identity Provider must be imported using federation_settings_id-idp_id, e.g.

Reviewer: ```suggestion
Identity Provider **must** be imported before using federation_settings_id-idp_id, e.g.
```

Reviewer: Still need

Author: Added to identity and org_config
